aatools/food_insecurity/fewsnet.py

```diff
@@ -0,0 +1,212 @@
+"""
+FEWS NET processing.
+
+Download and save the data provided by FEWS NET.
+"""
+
+import logging
+import zipfile
+from datetime import datetime
+from pathlib import Path
+
+from requests.exceptions import HTTPError
+
+from aatools.utils.io import download_url, unzip
+
+logger = logging.getLogger(__name__)
+
+
+def download_zip(
+    url: str,
+    zip_path: Path,
+    output_dir: Path,
+) -> bool:
+    """
+    Download and unzip the file at the url.
+
+    Parameters
+    ----------
+    url : str
+        url that contains the zip file to be downloaded
+    zip_path : Path
+        path to which the content of the url should be downloaded
+    output_dir : Path
+        path of dir to which the zip content should be written
+
+    Returns
+    -------
+    valid_file: bool
+        if True, the url contains a valid zip file
+
+    """
+    valid_file = False
+
+    try:
+        download_url(url, zip_path)
+        logger.info(f"Downloaded {url} to {zip_path}")
+
+        try:
+            unzip(zip_path, output_dir)
+            logger.debug(f"Unzipped {zip_path}")
+            valid_file = True
+        except zipfile.BadZipFile:
+            # indicates that the url returned something that wasn't a
+            # zip, happens often and indicates data for the given
+            # country - date is not available
+            logger.debug(
+                f"No zip data returned from url {url} "
+                f"check that the area and date exist."
+            )
+        # remove the zip file
+        zip_path.unlink()
+
+    except HTTPError:
+        logger.info(f"Couldn't download url {url}")
+
+    return valid_file
+
+
+def _download_fewsnet_country(
+    date: datetime,
+    iso2: str,
+    output_dir: Path,
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers the iso2 country.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code for which the data should be downloaded
+    output_dir: Path
+        path of dir to which the zip content should be written
+    use_cache: bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    country_data : bool
+        if True, country data for the given date and iso2 exists
+    """
+    base_url_country = "https://fdw.fews.net/api/ipcpackage/"
+    url_country = (
+        f"{base_url_country}?country_code={iso2}"
+        f"&collection_date={date.strftime('%Y-%m')}-01"
+    )
+
+    zip_path_country = output_dir / f"{iso2}{date.strftime('%Y%m')}.zip"
+    output_dir_country = output_dir / zip_path_country.stem
```

Comment:
What do you think of the following instead:
```python
output_dir_country = output_dir / f"{iso2}{date.strftime('%Y%m')}"
zip_path_country = Path(f"{output_dir_country}.zip")
```
That way you don't need to repeat `output_dir` which I found confusing, and it makes it clear that `zip_path_country` is the same as `output_dir_country` just with a `.zip` at the end

Reply:
Yeah that is what there was before, but then it coused confusion. But guess you right it is still better than what there is now. Will change it back